src/som/interpreter/actors/ReceivedMessage.java

```diff
@@ -85,9 +89,12 @@ public Object execute(final VirtualFrame frame) {
         dbg.prepareSteppingAfterNextRootNode();
       }
 
-      Object result = onReceive.call(frame, msg.args);
-
-      resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+      try {
+        Object result = onReceive.call(frame, msg.args);
+        resolvePromise(frame, msg.resolver, result, msg.triggerPromiseResolutionBreakpoint);
+      } catch (SomException exception) {
+        msg.getResolver().onError(exception);
+      }
```

Comment:
Same comments as above.
Might be useful to thing about how to remove that code duplication, too.